src/awkward1/operations/structure.py

```diff
@@ -487,8 +482,17 @@ def getfunction(inputs, depth):
             else:
                 return None
 
-        behavior = awkward1._util.behaviorof(base, what)
-        out = awkward1._util.broadcast_and_apply([base, what], getfunction, behavior)
+        keys = base.keys()
+        if where in base.keys():
+            keys.remove(where)
+        if len(keys) == 0:
+            # only key was removed, so just create new Record
+            behavior = awkward1._util.behaviorof(base, what)
+            out = (awkward1.layout.RecordArray([what], [where]),)
```

Comment:
not sure if in this case the behavior of `what` will be in the new array?

Reply:
I'll do a bigger write-up of this below, but the `behavior` is not supposed to be passed along here. It's only relevant for the final wrapper on the last line (`awkward._util.wrap` in the return value).